### PR TITLE
chore: update Yarn to 4.2.2 and remove yarn-*.cjs from the repo

### PR DESCRIPTION
**Pull request by HowardBraham**
## **Description**

- Updates Yarn to the latest version: 4.2.2
- Removes the `yarn-*.cjs` files from our repo (see this [discussion](https://github.com/yarnpkg/berry/issues/5748) about how these 2.6MB files shouldn't be committed)
- Now using `corepack enable` in CircleCI

This is very similar to MetaMask/metamask-module-template#243

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24930?quickstart=1)

## **Related issues**

Similar: MetaMask/metamask-module-template#243

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -354,6 +354,7 @@ jobs:
     executor: node-browsers-small
     steps:
       - run: *shallow-git-clone
+      - run: sudo corepack enable
```
Comment (legobeat) [inline .circleci/config.yml:357]: Does it with with just `corepack enable` sans sudo?

Comment (HowardBraham) [reply .circleci/config.yml:357]: Nope, this is what happens without sudo https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/84179/workflows/a00c10d8-5c25-4604-9332-5a71d51e512f/jobs/3025527

**Review by legobeat — DISMISSED** [review]

**Review by mcmire — DISMISSED** [review]
Nice!

**Review by legobeat — DISMISSED** [review]

**Review by mcmire — APPROVED** [review]

**Review by legobeat — APPROVED** [review]

**Comment by legobeat** [comment]
@HowardBraham This broke some GitHub Actions workflows.. 
Fixed in:
- #24973

**Pull request merged**